### PR TITLE
Launch Exam on TSB if TSB enabled

### LINKED ISSUE #35: Feature | Moodle-Quizaccess_Proctor | Launch Exam on TSB if TSB enabled
**Expected behaviour**: If tsb is enabled for the quizzes then launch the quiz in TSB browser

Solution: Extend the quizaccess rule base class of moodle and impement tsb launch on the attempt page setup.

**Testing Procdeures:**
1. create a quiz
2. select proctoring type
3. enable TSB 

Image

### PR DESCRIPTION
**Pull request by rohansharmasitoula**
## Description
- Launch Exam on TSB if TSB enabled

## Github Issue
- Resolves #35 

## Checklist before requesting a review
- [x] One line description of the changes is added in the PR
- [x] Issue is linked to the PR via commits (eg: resolves #123)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires only a documentation update

## Dependencies (if any):
- Mention any dependencies/issues for reviewers reference (eg: #1124 needs to be deployed before/after this PR)

## References (if any):
- [ ] Any reference documents to review before/after this PR review
- Add the links of any related PR/documents/diagrams for reviewers reference (eg: #link to understand login flow).

## Summary by CodeRabbit

- **New Features**
  - Enhanced quiz proctoring settings and access management.
  - Introduced compatibility with Talview Secure Browser (TSB).

- **Improvements**
  - Updated plugin version to 1.3.0 with corresponding build information.

- **Refactor**
  - Improved method signatures and bodies for better clarity and functionality.

- **Documentation**
  - Release notes updated to reflect new version and features.

### CODE REVIEW

File: rule.php [hunk 1]
```diff
@@ -227,32 +234,56 @@ public static function get_settings_sql($quizid) : array {
...
+    public function current_attempt_finished()
+    {
         $this->accessmanager->clear_session_access();
     }
 
+    public function is_tsb_required()
+    {
+        $quizaccess_proctor_setting_enabled = get_config('quizaccess_proctor', 'enableproctor');
+        $tsbenabled = $this->quizobj->get_quiz()->tsbenabled;
+        return ($quizaccess_proctor_setting_enabled && $tsbenabled);
```
Comment (devang1281) [inline rule.php:272]: @rohansharmasitoula Also add a check for `$proctortype !== 'noproctor'`

**Review by devang1281 — CHANGES_REQUESTED** [review]

File: rule.php (hunk 1 above)
Comment (rohansharmasitoula) [reply rule.php]: Added @devang1281

File: rule.php [hunk 2]
```diff
@@ -227,32 +234,59 @@ public static function get_settings_sql($quizid) : array {
...
+        return ($quizaccess_proctor_setting_enabled && $tsbenabled && $proctortype !== 'noproctor');
+    }
 
+    public function redirect_to_tsb_link()
+    {
+        global $PAGE, $CFG;
+        $baseUrl = $CFG->wwwroot;
+        $path = "/mod/quiz/view.php?id=" . $PAGE->cm->id;
+        $redirectURL = $baseUrl . $path;
+        $tsblink = "https://pages.talview.com/tsb/index.html?redirect_url=" . $redirectURL . "&user=" . $_SERVER ['HTTP_USER_AGENT'];
```
[the remark this answers is not in the document]
Comment (devang1281) [reply rule.php:282]: @rohansharmasitoula check this once

[the remark this answers is not in the document]
Comment (rohansharmasitoula) [reply rule.php]: > @rohansharmasitoula check this once

@devang1281 updated

**Review by devang1281 — APPROVED** [review]
LGTM

**Pull request merged**